tests/js/realm-tests.js

```diff
@@ -1673,37 +1673,112 @@ module.exports = {
 
     testDecimal128: function() {
         const realm = new Realm({schema: [schemas.DecimalObject]});
+        realm.write(() => { // FIXME: This should not be required!
+            realm.deleteAll();
+        });
 
-        let d = Decimal128.fromString("42");
-        realm.write(() => {
-            realm.create(schemas.DecimalObject.name, { decimalCol: d});
+        let numbers = [42, 3.1415, 6.022e23, -7, -100.2, 1.02E9];
+
+        numbers.forEach(number => {
+            let d = Decimal128.fromString(number.toString());
+            realm.write(() => {
+                realm.create(schemas.DecimalObject.name, { decimalCol: d});
+            });
         });
 
         let objects = realm.objects(schemas.DecimalObject.name);
-        TestCase.assertEqual(objects.length, 1);
+        TestCase.assertEqual(objects.length, numbers.length);
+
+        for (let i = 0; i < numbers.length; i++) {
+            let d128 = objects[i]['decimalCol'];
+            TestCase.assertTrue(d128 instanceof Decimal128);
+            TestCase.assertEqual(d128.toString(), numbers[i].toString().toUpperCase());
+        }
+
+        realm.close();
+    },
+
+    testDecimal128_LargeNumbers: function() {
+        const realm = new Realm({schema: [schemas.DecimalObject]});
+        realm.write(() => { // FIXME: This should not be required!
+            realm.deleteAll();
+        });
+        let numbers = ["1.02e6102", "-1.02e6102", "1.02e-6102", "9.99e6143", "1e-6142"];
+
+        numbers.forEach(number => {
+            let d = Decimal128.fromString(number);
+            realm.write(() => {
+                realm.create(schemas.DecimalObject.name, { decimalCol: d});
+            });
+        });
+
+        let objects = realm.objects(schemas.DecimalObject.name);
+        TestCase.assertEqual(objects.length, numbers.length);
 
-        let d128 = objects[0]['decimalCol'];
-        TestCase.assertTrue(d128 instanceof Decimal128);
-        TestCase.assertEqual(d128.toString(), "42");
+        for (let i = 0; i < numbers.length; i++) {
+            let d128 = objects[i]['decimalCol'];
+            TestCase.assertTrue(d128 instanceof Decimal128);
+            TestCase.assertEqual(d128.toString(), numbers[i].toUpperCase());
```

Comment:
it will be really nice if we could add an `equals(x: string): boolean` to all decimals128 values which does the comparison with the correct casing. I looked at https://github.com/mongodb/js-bson/blob/master/lib/decimal128.js and strangely enough there is no such thing 
Do you think we can add such method in this PR?

Reply:
envisioning this as  `d128.equals(string)`